packages/commonwealth/server/routes/updateChainCategory.ts

```diff
@@ -1,53 +1,51 @@
 import { AppError } from 'common-common/src/errors';
 import type { NextFunction, Response } from 'express';
-import type { ChainCategoryInstance } from 'server/models/chain_category';
 import type { DB } from '../models';
-import type { TypedRequestBody } from '../types';
+import type { TypedRequestBody, TypedResponse } from '../types';
 import { success } from '../types';
+import { ChainCategoryType } from 'common-common/src/types';
 
-type UpdateChainCategoryReq = Omit<ChainCategoryInstance, 'id'> & {
-  create: string;
+type UpdateChainCategoryReq = {
+  category: string;
+  chain_id: string;
   auth: string;
   jwt: string;
 };
 
+type UpdateChainCategoryRes = {
+  chainCategoryMap: { [chain: string]: ChainCategoryType[] };
+};
+
 const updateChainCategory = async (
   models: DB,
   req: TypedRequestBody<UpdateChainCategoryReq>,
-  res: Response,
+  res: TypedResponse<UpdateChainCategoryRes>,
   next: NextFunction
 ) => {
-  if (req.body.create === 'true') {
-    const categoryEntry = await models.ChainCategory.findOne({
+  if (Object.keys(ChainCategoryType).includes(req.body.category)) {
+    const chain = await models.Chain.findOne({
       where: {
-        chain_id: req.body.chain_id,
-        category_type_id: req.body.category_type_id,
+        id: req.body.chain_id,
       },
     });
-    if (!categoryEntry) {
-      const category = await models.ChainCategory.create({
-        chain_id: req.body.chain_id,
-        category_type_id: req.body.category_type_id,
-      });
-      return success(res, category.toJSON());
-    }
+    if (!chain) return next(new AppError('Invalid Chain Id'));
```

Comment:
Don't return next -- you can simply throw the AppError.

Reply:
✅